### PR TITLE
Metadata

### PR DESCRIPTION
**Pull request by pivnicek**
- [X] accept metadata from table manifest
- [X] test table manifest structure
- [X] write metadata from manifest
- [X] test writing metadata from manifest
- [X] write default system metadata
- [X] test writing default system metadata

Review needed

### CODE REVIEW

File: Docker/StorageApi/Writer.php [hunk 1]
```diff
@@ -359,6 +380,15 @@ public function uploadTables($source, $configuration = [])
                 The following unset should be removed once the 'escaped_by' parameter is removed from table manifest. */
                 unset($config['escaped_by']);
                 $config["primary_key"] = self::normalizePrimaryKey($config["primary_key"]);
+                if ($config["primary_key"] == [""]) {
```
Comment (odinuv) [inline Docker/StorageApi/Writer.php]: @pivnicek any idea where this came from?

Comment (pivnicek) [inline Docker/StorageApi/Writer.php]: @odinuv that is a strange one,  it's obviously a hack to deal with that `[""]` so I must have encountered that somewhere, but I can't remember.  It probably won't happen anymore sense you guys upgraded the PK handling though.  regardless that `==` makes me cry :)

Comment (odinuv) [inline Docker/StorageApi/Writer.php]: i thought so, @ondrejhlavacek can we remove this?

Comment (ondrejhlavacek) [inline Docker/StorageApi/Writer.php]: yup the `self::normalizePrimaryKey` method takes care of this situation

**Comment by pivnicek** [comment]
holy shit

**Pull request merged**